### PR TITLE
Support comments in workspace.josh files

### PR DESCRIPTION
**Pull request by LMG**
They start with # and are only in the first lines of the file (optionally)

### CODE REVIEW

File: src/filter/mod.rs [hunk 1]
```diff
@@ -538,23 +539,26 @@ fn unapply2<'a>(
...
+            let mapped = &tree::get_blob(&transaction.repo(), &tree, &Path::new("workspace.josh"));
+            let parsed = parse(mapped)?;
+
+            let mut blob = String::new();
+            if let Ok(c) = get_comments(mapped) {
+                if c.len() > 0 {
+                    blob = format!("{}", c);
+                }
+            }
+            let blob = &format!("{}{}\n", &blob, pretty(parsed, 0));
```
Comment (christian-schilling) [inline src/filter/mod.rs:551]: Will there allways be a newline at the end of the comments, or would this better be `{}\n{\n}\n`?

File: tests/proxy/workspace_invalid_trailing_slash.t [hunk 2]
```diff
@@ -134,6 +134,13 @@
   remote: Can't apply "mod workspace" (4e531443c5533e6d1b2503d0fad238cfc8491807)
   remote: Invalid workspace:
   remote: ----
+  remote:  --> 1:1
+  remote:   |
+  remote: 1 | a/ = :/sub1\xe2\x90\x8a (esc)
```
Comment (christian-schilling) [inline tests/proxy/workspace_invalid_trailing_slash.t:139]: Where do those extra cryptic characters at the end come from, and can we get rid of them?

File: src/filter/mod.rs (hunk 1 above)
Comment (LMG) [reply src/filter/mod.rs:551]: There will, because of the grammar, I think.

Comment (LMG) [reply src/filter/mod.rs:551]: Also all the newlines are kept as part of the comments. If we add one here, each push will increase the number of newlines by one.

**Review by christian-schilling — APPROVED** [review]

**Pull request merged**